source/grid/manifold_lib.cc

```diff
@@ -23,6 +23,119 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+
+
+namespace
+{
+  template <int dim1, int dim2>
+  Point<dim1+dim2> combine_points(const Point<dim1> &p1, const Point<dim2> &p2)
```

Comment:
Oh, I didn't pay attention -- this would only hold if you had this in the .h file. You don't here, so ignore my suggestion, unless you want to move everything to the header file as previously discussed.